mis_builder/models/mis_report_style.py

```diff
@@ -274,7 +274,8 @@ def to_xlsx_style(self, type, props, no_indent=False):
             if props.dp:
                 num_format += u'.'
                 num_format += u'0' * props.dp
-            num_format += '%'
+            # A comparison TYPE_PCT can keep its suffix
+            num_format += props.suffix if props.keep_suffix else '%'
```

Comment:
I'm not too comfortable with this `keep_suffix`.

Fundamentally, the root cause of this bug is that the comparison result is not of TYPE_PCT but is rather TYPE_NUM. 
The type of the comparison is different than the type of the compared values. This is the case when comparing two TYPE_NUM in percentage mode too, and it looks like what's in place for that can be considered as a workaround too.

So it think would be overall cleaner to:
- adapt `compare_and_render` to return the type of the comparison result as a fourth return value (the TYPE_NUM path can be adapted too)
- add a type property to `KpiMatrixCell`
- use that cell type property in priority over row.kpi.type if it is set when rendering

It's a bigger change but I hope the result should be much easier to understand.